examples/atari/dqn/train_dqn.py

```diff
@@ -169,22 +135,45 @@ def phi(x):
         eval_stats = experiments.eval_performance(
             env=eval_env,
             agent=agent,
-            n_steps=None,
-            n_episodes=eval_n_episodes)
-        print('n_runs: {} mean: {} median: {} stdev {}'.format(
-            eval_n_episodes, eval_stats['mean'], eval_stats['median'],
+            n_steps=args.eval_n_steps,
+            n_episodes=None)
+        print('n_episodes: {} mean: {} median: {} stdev {}'.format(
+            eval_stats['episodes'],
+            eval_stats['mean'],
+            eval_stats['median'],
             eval_stats['stdev']))
     else:
         experiments.train_agent_with_evaluation(
             agent=agent, env=env, steps=args.steps,
-            eval_n_steps=eval_n_steps,
-            eval_n_episodes=eval_n_episodes,
+            eval_n_steps=args.eval_n_steps,
+            eval_n_episodes=None,
             eval_interval=args.eval_interval,
             outdir=args.outdir,
-            save_best_so_far_agent=False,
+            save_best_so_far_agent=True,
             eval_env=eval_env,
         )
 
+        dir_of_best_network = os.path.join(args.outdir, "best")
+        agent.load(dir_of_best_network)
+
+        # Change seed for final evaluation
+        final_seed = max(train_seed, test_seed) - \
+            min(train_seed, test_seed) - 1
```

Comment:
Can you elaborate this calculation of `final_seed`?

Reply:
I'm trying to ensure that the train, test, and the final seed are different. We know that train and test seeds are different. I'm trying to get a third number that's different from the train and test seed. This is supposed to do that... Not sure it's the best way though. Ideally we do something like max(seeds) - min(seeds) to get a unique nonnegative seed.  However, if one of the seeds is 0, then max(seeds) - min(seeds) = max(seeds), and we'll have final_seed= max_seed. This ensures that doesn't happen.

Reply:
The algorithm does not work if seeds=[1,3], and it can produce a negative value.

If we use `eval_env` for both regular and final evaluation, I think it is ok to use `eval_env` for final evaluation without without additionally calling `eval_env.seed(final_seed)`, since its internal random state is already changed.